sky/jobs/server/core.py

```diff
@@ -367,107 +366,155 @@ def launch(
     modified_catalogs = {} if consolidation_mode_job_ids is not None else (
         service_catalog_common.get_modified_catalog_file_mounts())
 
+    # For pools with num_jobs > 1, create job IDs upfront with one task.
+    pre_created_job_ids: Optional[List[int]] = None
+    if (pool is not None and num_jobs > 1 and
+            consolidation_mode_job_ids is None):
+        with rich_utils.safe_status(
+                ux_utils.spinner_message('Creating job IDs')):
+            # Get controller handle to create job IDs upfront
+            try:
+                local_handle = backend_utils.is_controller_accessible(
+                    controller=controller, stopped_message='')
+                backend = backend_utils.get_backend_from_handle(local_handle)
+                assert isinstance(backend, backends.CloudVmRayBackend)
+
+                # Create num_jobs - 1 job IDs upfront (the controller task's ray
+                # job ID will be used as the nth job ID)
+                resources_str = backend_utils.get_task_resources_str(
+                    dag.tasks[0], is_managed_job=True)
+                job_name = dag.name
+                result = backend.add_job(handle=local_handle,
+                                         job_name=job_name,
+                                         resources_str=resources_str,
+                                         metadata=dag.tasks[0].metadata_json,
+                                         num_jobs=num_jobs - 1)
```

Comment:
By manually calling this we will have a bunch of jobs on the remote jobs controller with status permanently stuck at INIT. That's not ideal but it's fine - we shouldn't be using the underlying ray job status for anything since #4485.
We should make sure that it won't block the cluster autostop though.